src/transformers/trainer.py

```diff
@@ -509,11 +509,18 @@ def _get_train_sampler(self) -> Optional[torch.utils.data.sampler.Sampler]:
 
         # Build the sampler.
         if self.args.group_by_length:
+            model_input_name = self.tokenizer.model_input_names[0] if self.tokenizer is not None else None
```

Comment:
@sgugger - the `LengthGroupedSampler` is very important to speed-up training for Wav2Vec2, but the processor has `input_values` as its model input name so that we have to pass this to the `LengthGroupedSampler`. See changes in `src/transformers/trainer_pt_utils.py`